### PR TITLE
feature: add provisioning for storage-account FileService

### PR DESCRIPTION
**Pull request by Meir017**

### CODE REVIEW

**Comment by Meir017** [comment]
ran the new tests using the following:

```ps1
$token = (az account get-access-token --resource=https://management.core.windows.net/ -o json | ConvertFrom-Json).accessToken
$env:SUBSCRIPTION_ID = 'caedc0f5-7aa8-4612-98f7-9ce6ad4dc1ba'
$env:AZURE_TEST_MODE='Record'
$env:TEST_CSM_ORGID_AUTHENTICATION = "RawToken=$token"
```

File: sdk/provisioning/Azure.Provisioning.Storage/tests/StorageTests.cs [hunk 1]
```diff
@@ -39,6 +39,26 @@ public async Task StorageBlobDefaults()
             await ValidateBicepAsync();
         }
 
+        [RecordedTest]
+        public async Task StorageFileDefaults()
```
Comment (JoshLove-msft) [inline sdk/provisioning/Azure.Provisioning.Storage/tests/StorageTests.cs:43]: Can you also update the ExistingResources test to cover the new resource?

Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning.Storage/tests/StorageTests.cs:43]: You will need to push the new test recordings - https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#pushing-session-records-and-updating-assetsjson

Comment (Meir017) [reply sdk/provisioning/Azure.Provisioning.Storage/tests/StorageTests.cs:43]: done

**Comment by Meir017** [comment]
@JoshLove-msft can you review again?

**Pull request closed without merging**